src/languageserver/handlers/settingsHandlers.ts

```diff
@@ -88,7 +88,9 @@ export class SettingsHandler {
       if (settings.files?.associations) {
         for (const [ext, languageId] of Object.entries(settings.files.associations)) {
           if (languageId === 'yaml') {
-            this.yamlSettings.fileExtensions.push(ext);
+            const index = ext.indexOf('*');
```

Comment:
A quick search through the code base shows that there is already a utility for matching file paths to globs (but only very simple globs); the `FilePatternAssociation` class.

https://github.com/redhat-developer/yaml-language-server/blob/f7407e2205f7611835e0cc61a00ccfa653e41f34/src/languageservice/services/yamlSchemaService.ts#L72

Example usage:

``` ts
const fpa = new FilePatternAssociation(globPath);
const isMatch = fpa.matchesPattern(myFilePath);
```

A closer look shows it only really handles the wildcard `*` globs and would fail on other syntax such as `?` etc.

But it may be better than nothing, with the potential to add support for more glob syntax in the future?